### PR TITLE
UF-250 - Restructure context menus

### PR DESCRIPTION
**Pull request by cristianonicolai**
Removed context menus from main nav bar when using standard navigation.
Context menus moved into a tertiary menu bar.

### CODE REVIEW

File: uberfire-workbench/uberfire-workbench-client-views-bs2/src/main/java/org/uberfire/client/views/bs2/menu/PerspectiveContextMenusView.java [hunk 1]
```diff
@@ -1,136 +0,0 @@
...
-        }
-
-        return null;
-    }
-
-    Widget makeMenuItemCommand( final MenuItem item ) {
-        final MenuItemCommand cmdItem = (MenuItemCommand) item;
-        final Widget gwtItem;
-
-        gwtItem = new NavLink( cmdItem.getCaption() ) {{
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-bs2/src/main/java/org/uberfire/client/views/bs2/menu/PerspectiveContextMenusView.java]: GWT.create?

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/PerspectiveContextMenusView.java [hunk 2]
```diff
@@ -1,107 +0,0 @@
...
-
-    @Override
-    public Widget asWidget() {
-        return null;
-    }
-
-    @Override
-    public void buildMenu( final Menus menus ) {
-        clear();
-        menus.accept( new AuthFilterMenuVisitor( authzManager, identity, new StackedDropdownMenuVisitor( new HasMenuItems() {
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/PerspectiveContextMenusView.java]: This logic should be moved to presenter.

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuCompactNavBarView.java [hunk 3]
```diff
@@ -81,9 +70,25 @@ public void addMenus( final Menus menus ) {
...
-    public void selectMenu( final MenuItem menu ) {
-        selectElement( menu.getCaption(), listItemMap.get( menu ) );
+    public void selectMenu( final MenuItem menu, final Menus context ) {
+        selectElement( menu.getCaption(), getListItemMap().get( menu ) );
+
+        removeContextMenuWidgets( navbarNav );
+
+        if ( context == null || context.getItems().isEmpty() ) return;
+
+        context.accept( new AuthFilterMenuVisitor( authzManager, identity, new StackedDropdownMenuVisitor( new HasMenuItems() {
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuCompactNavBarView.java]: Logic on View? isn't better to move this to presenter?

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuNavBarView.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                final Widget child = widgets.next();
+                if ( child.getElement().hasClassName( UF_PERSPECTIVE_CONTEXT_MENU ) ) {
+                    widgets.remove();
+                }
+            }
+        }
+    }
+
+    public void clear() {
+        GWT.log( "clear" );
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuNavBarView.java]: remove this log?

**Comment by ederign** [comment]
@cristianonicolai there is a special reason to have this logic on view?

btw: it's working on showcase (old one/new one) http://prntscr.com/96l7g3

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuCompactNavBarView.java [hunk 5]
```diff
@@ -81,9 +70,25 @@ public void addMenus( final Menus menus ) {
         menus.accept( new AuthFilterMenuVisitor( authzManager, identity, new CompactMenuVisitor( this ) ) );
     }
 
-    @Override
-    public void selectMenu( final MenuItem menu ) {
-        selectElement( menu.getCaption(), listItemMap.get( menu ) );
+    public void selectMenu( final MenuItem menu, final Menus context ) {
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuCompactNavBarView.java]: logic on view

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuNavBarView.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.Widget;
+import org.gwtbootstrap3.client.ui.NavbarNav;
+import org.gwtbootstrap3.client.ui.base.ComplexWidget;
+import org.jboss.errai.security.shared.api.identity.User;
+import org.uberfire.security.authz.AuthorizationManager;
+import org.uberfire.workbench.model.menu.MenuItemPerspective;
+
+/**
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuNavBarView.java]: remove this.

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java [hunk 7]
```diff
@@ -41,22 +37,12 @@
  * Created by Cristiano Nicolai.
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java]: also this. probably is on your IDEA default.

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuCompactNavBarView.java [hunk 8]
```diff
@@ -46,29 +42,22 @@
  * Created by Cristiano Nicolai.
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuCompactNavBarView.java]: remove.

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuNavBarView.java (hunk 4 above)
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuNavBarView.java]: remove.

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java [hunk 9]
```diff
@@ -71,36 +57,67 @@ public void addMenuItem( MenuPosition position, final Widget menuContent ) {
...
+                this.navbarNav.add( menuContent );
                 break;
         }
     }
 
     @Override
-    public void selectMenu( final MenuItem menu ) {
-        selectElement( listItemMap.get( menu ) );
+    public void selectMenu( final MenuItem menu, final Menus context ) {
+        final Widget item = getListItemMap().get( menu );
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java]: More logic on view.

**Comment by cristianonicolai** [comment]
Hi @ederign, I' ve updated this PR based on your comments, please have a look once you got a chance ;)

**Comment by ederign** [comment]
@cristianonicolai just need to recheck if the empty methods are ok! If so, this PR has my +1.

I've already tested on master.

**Comment by cristianonicolai** [comment]
@ederign those methods are intentionally empty yes. They're the BS2 module which we're not maintaining on master.  My suggestion would be for us to remove the entire module as I don' t really see any point on keeping it if we don't plan to maintain/improve. @porcelli @manstis @csadilek what do you guys think?

**Comment by manstis** [comment]
@cristianonicolai @ederign @csadilek @porcelli I'd favour removing the BS2 module if we're to accept "empty" implementations for changes made to the underlying interfaces/API. It'll remain in the history should we ever need it again...

**Comment by csadilek** [comment]
+1, and I agree with removing the BS2 module.

**Comment by csadilek** [comment]
Thanks.

master: 79bd9bf984f4511b45fc67921c6a45e4a3eacd43

**Pull request closed without merging**